prov/mxm/src/mlxm.h

```diff
@@ -0,0 +1,306 @@
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * OpenFabrics.org BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *	- Redistributions of source code must retain the above
+ *	  copyright notice, this list of conditions and the following
+ *	  disclaimer.
+ *
+ *	- Redistributions in binary form must reproduce the above
+ *	  copyright notice, this list of conditions and the following
+ *	  disclaimer in the documentation and/or other materials
+ *	  provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _FI_MLXM_H
+#define _FI_MLXM_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#if HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <errno.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <pthread.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netdb.h>
+#include <rdma/fabric.h>
+#include <rdma/fi_prov.h>
+#include <rdma/fi_domain.h>
+#include <rdma/fi_endpoint.h>
+#include <rdma/fi_tagged.h>
+#include <rdma/fi_rma.h>
+#include <rdma/fi_cm.h>
+#include <rdma/fi_errno.h>
+#include <rdma/fi_log.h>
+#include <fi_list.h>
+#include <mxm/api/mxm_api.h>
+#include "mpool.h"
+#include "uthash.h"
+static int mlxm_errno_table[MXM_ERR_LAST] = {
+	0,			/* MXM_OK = 0, */
+	-FI_ENOMSG,		/* MXM_ERR_NO_MESSAGE, */
+	-EWOULDBLOCK,		/* MXM_ERR_WOULD_BLOCK, */
+	-FI_EIO,		/* MXM_ERR_IO_ERROR, */
+	-FI_ENOMEM,		/* MXM_ERR_NO_MEMORY, */
+	-FI_EINVAL,		/* MXM_ERR_INVALID_PARAM, */
+	-FI_ENETUNREACH,	/* MXM_ERR_UNREACHABLE, */
+	-FI_EINVAL,		/* MXM_ERR_INVALID_ADDR, */
+	-FI_ENOSYS,		/* MXM_ERR_NOT_IMPLEMENTED, */
+	-FI_EMSGSIZE,		/* MXM_ERR_MESSAGE_TRUNCATED, */
+	0,			/* MXM_ERR_NO_PROGRESS, */
+	-FI_ETOOSMALL,		/* MXM_ERR_BUFFER_TOO_SMALL, */
+	-FI_ENOENT,		/* MXM_ERR_NO_ELEM, */
+	-FI_EIO,		/* MXM_ERR_SOME_CONNECTS_FAILED, */
+	-FI_ENODEV,		/* MXM_ERR_NO_DEVICE, */
+	-FI_EBUSY,		/* MXM_ERR_BUSY, */
+	-FI_ECANCELED,		/* MXM_ERR_CANCELED, */
+	-FI_EINVAL,		/* MXM_ERR_SHMEM_SEGMENT, */
+	-EEXIST,		/* MXM_ERR_ALREADY_EXISTS, */
+	-FI_EINVAL,		/* MXM_ERR_OUT_OF_RANGE, */
+	-FI_ETIMEDOUT		/* MXM_ERR_TIMED_OUT, */
```

Comment:
Maybe use:
[MXM_ERR_NO_MESSAGE] = -FI_ENOMSG
to ensure that values are matched properly